src/proofnetwork/restclient.cpp

```diff
@@ -655,6 +739,16 @@ QPair<QString, QString> RestClientPrivate::parseHost(const QString &host)
     return {newHost, postfix};
 }
 
+void RestClientPrivate::sendMailAboutSlowNetwork(QNetworkReply *reply, long timeout)
+{
+    auto ips = reply->request().rawHeader("Proof-IP-Addresses");
```

Comment:
I would avoid relying on this header due to possible problems in future if we decide to alter or discard it